### DISCUSSION TITLE
Problem with zarr metadata when using zarr memory store

### DISCUSSION THREAD

**Discussion by joaobuibergen** [Parcels-code/Parcels#1460] [Q&A]
Hi,

I've reluctantly moved to the new version of parcels with zarr output and have been plowing through a series of errors related to the zarr format. The most recent is the following, related to some missing zmetadata dictionary key. I've looked around and it seems it is related to unconsolidated metadata. What should I do?

Can anyone help me with this?

compute FSLE for 2005-01-01 12:00:00
 at 0 days, 0 hours, 0 minutes, 0 seconds. 
INFO: Compiled ArrayJITParticleAdvectionRK4 ==> /tmp/parcels-153555/lib758736f00c74a81306858c5e1c8c8281_0.so
INFO: Output files are stored in <zarr.storage.MemoryStore object at 0x7f4e63848d90>.zarr.
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████| 15552000.0/15552000.0 [03:08<00:00, 82650.43it/s]
Traceback (most recent call last):
  File "/home/johor1356/.conda/envs/py3_parcels/lib/python3.9/site-packages/zarr/storage.py", line 826, in __getitem__
    value = parent[key]
KeyError: '.zmetadata'

**Comment by erikvansebille** [comment]
Thank you for reporting @joaobuibergen, I'd be happy to look into this. But in order to help you, we need (much) more information. What is the code you ran that created this error?

**Comment by joaobuibergen** [comment]
Hi Erik,

I've made further debugging. The error seems to occur randomly, so I'm
suspecting it could be a hardware issue. I'll try to verify this before
pursuing this further.

João Bettencourt
SEAS Fellow
Geophysical Institute - University of Bergen
Bjerknes Centre for Climate Research
https://www.uib.no/en/persons/João.Bettencourt
